addons/mail/models/mail_activity.py

```diff
@@ -319,23 +316,13 @@ def write(self, values):
             # send bus notifications
             todo_activities = user_changes.filtered(lambda act: act.date_deadline <= fields.Date.today())
             if todo_activities:
-                self.env['bus.bus']._sendmany([
-                    [partner, 'mail.activity/updated', {'activity_created': True}]
-                    for partner in todo_activities.user_id.partner_id
-                ])
-                self.env['bus.bus']._sendmany([
-                    [partner, 'mail.activity/updated', {'activity_deleted': True}]
-                    for partner in pre_responsibles
-                ])
+                (todo_activities.user_id + pre_responsibles)._bus_send('mail.activity/updated', {'activity_created': True})
         return res
 
     def unlink(self):
         todo_activities = self.filtered(lambda act: act.date_deadline <= fields.Date.today())
         if todo_activities:
-            self.env['bus.bus']._sendmany([
-                [partner, 'mail.activity/updated', {'activity_deleted': True}]
-                for partner in todo_activities.user_id.partner_id
-            ])
+            todo_activities.user_id._bus_send('mail.activity/updated', {'activity_deleted': True})
```

Comment:
`_bus_channel` on activity too?
(there are plans some day to make this better, expand it, and sync chatter too)

Reply:
Sending to the activity should be sending to the assigned user, sending to the current user (to reflect the change in all tabs), and to all users currently viewing the chatter of the related record (feature to come at some point in the future).

So it's more complicated/can be ignored for now. These activity notif are not very well thought out at the moment anyway.